### ISSUE TITLE
Socket timeout when resetting source tracking

### ISSUE THREAD

**Issue by guillaumezeeman** [forcedotcom/cli#2269]
### Summary
When I try to reset the source tracking of my scratch org using the 'sfdx project reset tracking' SFDX command, I will always get an socket connection timeout. This command will fail with the following error message:

-----------------------------------------------------------------------------
Error (1): request to https://inspiration-app-6208-dev-ed.my.salesforce.com/services/data/v58.0/tooling/query/0r8xx1rSAGmGcutA2C-18000 failed, reason: Socket connection timeout

-----------------------------------------------------------------------------
It seems to timeout when retrieving the status of the files from the org using a REST API call. I've tried several things to try and fix this issue, all to no avail. 

I have tried to:
- First do a 'sfdx project reset tracking' before resetting the tracking
- Reverting to an older version of SFDX CLI
- Using the old 'sfdx force:source:tracking:reset' command
- Creating a new scratch org and trying to reset my source code with that one
- Try resetting my source tracking using a different wifi netwerk
- Use 'sfdx doctor' to retrieve log files

### Steps To Reproduce
- Create a new scratch org
- Update your scratch org
- Switch to different branch
- Reset your source tracking with the following command
sfdx project reset tracking -po {alias}

I do not have a repo that I can share, as the problem occurs with an application of a private company and the source code is not open source.

### Expected result
The source tracking as been synced with the current files

### Actual result
The command fails with a socket connection timeout

### System Information
**System:** MacOS Ventura 13.0.1 (22A400)
**SFDX version:** sfdx-cli/7.206.6 darwin-arm64 node-v20.2.0
**Terminal:** zsh

```json
{
  "code": 1,
  "context": "ResetTracking",
  "commandName": "ResetTracking",
  "message": "Nonexistent flag: --verbose\nSee more help with --help",
  "name": "Error",
  "status": 1,
  "stack": "Error: Nonexistent flag: --verbose\nSee more help with --help\n    at validateArgs (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/parser/validate.js:9:19)\n    at validate (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/parser/validate.js:157:5)\n    at Object.parse (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/parser/index.js:19:35)\n    at async ResetTracking.parse (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/command.js:221:25)\n    at async ResetTracking.run (/usr/local/lib/node_modules/sfdx-cli/node_modules/@salesforce/plugin-deploy-retrieve/src/commands/project/reset/tracking.ts:54:23)\n    at async ResetTracking._run (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/command.js:117:22)\n    at async Config.runCommand (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/config/config.js:329:25)\n    at async run (/usr/local/lib/node_modules/sfdx-cli/node_modules/@oclif/core/lib/main.js:89:16)",
  "exitCode": 1,
  "warnings": []
}
```

### Additional information
None.

**Comment by guillaumezeeman** [comment]
I have discovered a workaround that fixes this particular problem. If I deploy a specific file to my scratch org using the push command (like '**sfdx project deploy start -m CustomObject:MyCustomObject__c**') and then try to reset my source tracking, I do not receive an timeout anymore. 

My guess is that SFDX does something to the status of the source tracking which revolves the timeout issue. This does not fix the initial problem, because I will still get this issue with a new scratch org, but at least I have a way to reset my source tracking again.

**Comment by JonnyPower** [comment]
Something similar here is breaking our CI due to exorbitant run times, where the environment variables that used to skip source tracking are failing and now CI spends hours spinning post deployment. Wonder if same root cause.

**Comment by amtrack** [comment]
We're experiencing the same thing as @JonnyPower .
`SFDX_SOURCE_MEMBER_POLLING_TIMEOUT ` or `SF_SOURCE_MEMBER_POLLING_TIMEOUT ` don't seem to have an effect anymore.

**Comment by JonnyPower** [comment]
@amtrack we're seeing success / stability on `7.207.4` - hope upgrading resolves for you too

**Comment by amtrack** [comment]
@JonnyPower Yes, I can confirm that `7.207.5` seems to have fixed the issue for us, too.

**Comment by guillaumezeeman** [comment]
My issue also seems to be resolved with the `7.207.5` version.

**Comment by shetzel** [comment]
I believe this was related to the isomorphic-git issue reported here: https://github.com/forcedotcom/cli/issues/2194
This would explain why sfdx v7.207.5 fixed things.  Thanks for reporting and please reopen if you are still seeing this later.
